api/src/main/java/org/eclipse/microprofile/lra/annotation/Status.java

```diff
@@ -46,14 +46,6 @@
  * report <em>404 Not Found</em> HTTP status code. This enables the
  * participant to free up resources.
  *
- * If this status method is invoked before either of the <em>@Compensate</em>
- * or <em>@Complete</em> methods have been called then the participant
- * should report that it does not yet have a status using a JAX-RS
- * exception mapper that maps to a <em>412 Precondition Failed</em> HTTP
- * status code (such as
- * {@link org.eclipse.microprofile.lra.client.IllegalLRAStateException}
- * but any exception that maps to 412 will do).
- *
```

Comment:
I am approving PR but I just wanted to ask - is this covered somewhere else in the spec or do we leave this behavior unspecified?

Reply:
@xstefank Earlier in the project the state model for a participant was only entered via the `Compensating` or the `Completing` states. Later on we added a new state called `Active` which is entered when the participant joins the LRA. Though not strictly necessary, the new state had the advantage of simplying the spec and is more natural for the business code to return a status of Active instead of the exception together with a mapper that maps to 412 and this text should have been removed back then. One could argue that the extra state adds complexity but the benefits probably outweigh the drawbacks.

So to answer your question, the behaviour should be and is fully specified.